zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroAuthenticationService.java

```diff
@@ -247,6 +248,9 @@ public boolean isAuthenticated() {
         } else if (ACTIVE_DIRECTORY_GROUP_REALM.equals(name)) {
           allRoles = ((ActiveDirectoryGroupRealm) realm).getListRoles();
           break;
+        } else if (realm instanceof KnoxJwtRealm) {
```

Comment:
Not sure if the other realms need to do string checks or should be changed to instanceof checks, but had to do it here for the test using the spy

Reply:
You have an advantage when you do a string comparison.
There is a fixed string in the configuration and you can change the underlying implementation independently of the configuration.

Reply:
I don't really follow. We have to cast it to get the class specific behavior, so we're kinda relying on the implementation. And it prevents subclassing

Reply:
Forget my previous stupid comment. You are right. I didn't look at all the lines of code.